### PR TITLE
Document api endpoints

### PR DESCRIPTION
**Pull request by chotai**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,25 +45,186 @@ cd cdp-uploader
...
+
+#### Body parameters:
+
+| Parameter name        | Description                                                        | Required |
+| --------------------- | ------------------------------------------------------------------ | -------- |
+| redirect              | Url to redirect to after file has been successfully uploaded.      | yes      |
+| destinationBucket     | S3 bucket that file will be moved to once the scanning is complete | yes      |
+| destinationPath       | 'Folder' in bucket where scanned files will be placed              | no       |
+| scanResultCallbackUrl | Url that will be called once all files in upload have been scanned | no       |
+| metadata              | Map of additional information related to upload                    | no       |
```
Comment (flurdy) [inline README.md:81]: Will be added to the S3 file?

Comment (flurdy) [reply README.md:81]: Need to double check that...

File: README.md [hunk 2]
```diff
@@ -44,25 +45,186 @@ cd cdp-uploader
...
+| ----------- | ------------------------------------------------- | -------- |
+| User-Agent  | Identifier of the service that calls cdp-uploader | TBD      |
+
+#### Body parameters:
+
+| Parameter name        | Description                                                        | Required |
+| --------------------- | ------------------------------------------------------------------ | -------- |
+| redirect              | Url to redirect to after file has been successfully uploaded.      | yes      |
+| destinationBucket     | S3 bucket that file will be moved to once the scanning is complete | yes      |
+| destinationPath       | 'Folder' in bucket where scanned files will be placed              | no       |
```
Comment (flurdy) [inline README.md:79]: Otherwise root prefix

Comment (flurdy) [reply README.md:79]: I guess it is explained in the note below

File: README.md (hunk 1 above)
Comment (flurdy) [reply README.md:81]: Add to note perhaps

File: README.md [hunk 3]
```diff
@@ -44,25 +45,186 @@ cd cdp-uploader
...
+
+> [!NOTE]
+> We will generate an uploadId for the upload and fileIds for files in the upload request. Files (objects) will be moved to destination bucket under the path uploadId/fileId which will be prefixed with the bucket path if it has been provided.
+
+#### Example response
+
+```json
+{
+  "uploadId": "b18ceadb-afb1-4955-a70b-256bf94444d5",
+  "uploadAndScanUrl": "https://cdp-uploader/upload-and-scan/b18ceadb-afb1-4955-a70b-256bf94444d5",
```
Comment (flurdy) [inline README.md:91]: Relative?

File: README.md [hunk 4]
```diff
@@ -44,25 +45,186 @@ cd cdp-uploader
...
+## GET /status/{upload-id}
+
+The status API provides information about uploaded files, virus scan status and S3 location.
+The API is intended to be polled by the frontend services, it is not public and cannot be called direct from the browser.
+
+#### Request Parameters
+
+| Parameter Name | Description                                                          |
+| -------------- | -------------------------------------------------------------------- |
+| upload-id      | Unique id for that upload. ID is provided via the `/initialize` call |
```
Comment (flurdy) [inline README.md:133]: `/initiate`

File: README.md [hunk 5]
```diff
@@ -44,25 +45,186 @@ cd cdp-uploader
...
+      "contentLength": 82330,
+      "checksumSha256": "S/W84Awhf0KWN2aBkk+up52srTpiOCUHs/r1nHdJlB0=",
+      "filename": "unicorn.jpg",
+      "s3Bucket": "cdp-example-node-frontend",
+      "s3Key": "ba477d67-0005-4c26-a673-fa5139a2adf5/07aca302-fe02-4650-adf3-f1cdd8bc3988"
+    }
+  ],
+  "fields": {
+    "a-form-field": "some value",
+    "file-upload": {
```
Comment (flurdy) [inline README.md:159]: `file-upload-field` to make is clear it is a field name?

File: README.md (hunk 1 above)
Comment (chotai) [reply README.md:81]: We don't add to S3 metadata

Comment (flurdy) [reply README.md:81]: Ah no we dont, I thought we did. Nevermind.

**Review by flurdy — APPROVED** [review]

**Review by flurdy — APPROVED** [review]

**Pull request merged**